packages/manager/apps/pci-rancher/src/utils/rancher.ts

```diff
@@ -69,3 +75,90 @@ export const getRancherPlanDescription = (rancherPlan: RancherPlan['name']) => {
       return null;
   }
 };
+
+/**
+ * Extracts drivers and plan information from a switch plan error message.
+ *
+ * @param inputString - The input error message string.
+ * @returns - An object containing the drivers and plan, or null if extraction fails.
+ * */
+export function extractDriversAndPlanFromSwitchPlanError(
+  inputString: string,
+): null | { drivers: string[]; plan: string } {
+  const beginPhrase = 'Unable to switch to plan';
```

Comment:
Maybe use a regex for the whole string would be more readable.

Something like : ```^Unable to switch to plan (?<plan>[^\s:]+)\s?:[^[]+\[(?<drivers>[^]]+)]```

Reply:
yes i know but i had a lot of issue with regex with sonar

Reply:
Agree that it will be safer and will remove boilerplate code